resources/build/build-utils.sh

```diff
@@ -137,3 +137,39 @@ findTier
 findVersion
 # printVersionUtilsDebug
 printBuildNumberForTeamCity
+
+# Intended for use with macOS-based builds, as Xcode build phase "run script"s do not have access to important
+# environment variables.  Doesn't hurt to run it at other times as well.  The output file is .gitignore'd.
+function exportEnvironmentDefinitionScript() {
+    ENVIRONMENT_SH="$KEYMAN_ROOT/resources/environment.sh"
+
+    # Remove old copy if it exists
+    rm $ENVIRONMENT_SH > /dev/null 2> /dev/null 
+
+    # Documentation about the script, within the script.
+    echo "# Do not edit - this is an autogenerated script.  See build/build-utils.sh for more details." >> $ENVIRONMENT_SH
+    echo "# This file redefines critical environment variables for import to Xcode build phases." >> $ENVIRONMENT_SH
+    echo "" >> $ENVIRONMENT_SH
+
+    # Defining variables for VERSION here will leave static definitions that don't automatically update when a user
+    # changes branches; some branches are 'similar enough' to not require full command-line based rebuilds.
```

Comment:
When does this happen? On developer machines only I would think?

Reply:
"here" = "within `exportEnvironmentDefinitionScript`".

As far as when "changing branches", it wouldn't be an issue for CI because it always "starts fresh."  It'd only affect developer machines, yes.